src/otx/v2/cli/cli.py

```diff
@@ -349,8 +349,8 @@ def instantiate_classes(self, subcommand: str) -> None:
         self.model = self.auto_runner.get_model(model={**model_cfg}, num_classes=num_classes)
         # For prediction class
         # NOTE: We'll need to move this somewhere other than here in the next phase.
-        if num_classes is not None and "num_classes" in model_cfg.get("head", {}):
-            model_cfg["head"]["num_classes"] = num_classes
+        if getattr(self.model, "config_dict", None):
+            model_cfg = self.model.config_dict
```

Comment:
How about something like this to make it easier to follow?
```suggestion
       model_cfg = self.model.config_dict if hasattr(self.model, 'config_dict') else None
```

Reply:
I modified it, but someone may think it is a little bit awkward. Please check this.

Reply:
yeah, this does not look good. I think I misunderstand the code

Reply:
so we assign `self.model.config_dict` even if it is `None`?

Reply:
if `self.model.config_dict` is `None` then code will pass the if branch. So model_cfg will not be changed.

Reply:
thanks for the clarification. My bad, let's keep the old version